samcli/lib/init/__init__.py

```diff
@@ -145,3 +150,30 @@ def _enable_application_insights(application_insights: bool, output_dir: str, na
         template_modifier = ApplicationInsightsTemplateModifier(template_file_path)
         template_modifier.modify_template()
         EventTracker.track_event(EventName.USED_FEATURE.value, UsedFeature.INIT_WITH_APPLICATION_INSIGHTS.value)
+
+
+def _create_default_samconfig(package_type: str, output_dir: str, name: str) -> None:
+    """
+    Init post-processing function to create a default samconfig.toml
+    file and place it into the cookie cutter project.
+
+    Parameters
+    ----------
+    package_type: str
+        string representing the package type, 'Zip' or 'Image', see samcli/lib/utils/packagetype.py
+    output_dir: str
+        string representing the output directory of the template
+    name:
+        string representing the name of the application
+    """
+    # NOTE(sriram-mv): If this is coming from a `--location` without a corresponding project name,
+    # do not attempt to insert a samconfig.toml for a straight cookie-cutter init.
+    if not name:
+        return
+    project_root_path = Path(output_dir, name) if name else Path(output_dir)
+    if Path(project_root_path, "samconfig.toml").is_file():
```

Comment:
that is a fair call out, it can be changed.

Reply:
I thought about this some more. As long its not overwriting an existing file we should be okay. And if customers are using another .toml file they are already having to specify the that file since its no longer a default. Might be a good way for customers to learn about this and adopt some patterns if they choose to adopt it.